nixos/modules/security/pam.nix

```
@@ -1512,16 +1577,30 @@ in
           Did you forget to set `services.openssh.enable` ?
         '';
       }
+      {
+        assertion = with config.security.pam.rssh;
+          enable -> (settings.auth_key_file or null != null || settings.authorized_keys_command or null != null);
+        message = ''
+          security.pam.rssh.enable requires either security.pam.rssh.settings.auth_key_file or
+          security.pam.rssh.settings.authorized_keys_command to be set.
+        '';
+      }
     ];
 
     warnings = lib.optional
-      (with lib; with config.security.pam.sshAgentAuth;
+      (with config.security.pam.sshAgentAuth;
         enable && lib.any (s: lib.hasPrefix "%h" s || lib.hasPrefix "~" s) authorizedKeysFiles)
       ''config.security.pam.sshAgentAuth.authorizedKeysFiles contains files in the user's home directory.
 
         Specifying user-writeable files there result in an insecure configuration:
         a malicious process can then edit such an authorized_keys file and bypass the ssh-agent-based authentication.
         See https://github.com/NixOS/nixpkgs/issues/31611
+      '' ++ lib.optional
+      (with config.security.pam.rssh;
+        enable && settings.auth_key_file or null != null && settings.authorized_keys_command or null != null) ''
+        config.security.pam.rssh.settings.auth_key_file will be ignored as
+        pam.security.pam.rssh.settings.authorized_keys_command has been specified.
```

nit: typo

```suggestion
        config.security.pam.rssh.settings.authorized_keys_command has been specified.
```

Also, maybe we should strip the `config` everywhere in documentation and error messages, since it can be assumed as the options root.